AmplifyPlugins/Predictions/AWSPredictionsPlugin/Support/Utils/IdentifyTextResultTransformers.swift

```diff
@@ -229,32 +257,20 @@ class IdentifyTextResultTransformers: IdentifyResultTransformers {
             return nil
         }
 
+        // KEY_VALUE_SET block contains CHILD and VALUE entity type blocks
         for keyBlockRelationship in relationships {
-            guard let text = keyBlock.text,
-                let ids = keyBlockRelationship.ids else {
-                    continue
+
+            guard let ids = keyBlockRelationship.ids else {
+                continue
             }
+
             switch keyBlockRelationship.types {
-            case .child where keyBlock.blockType == .word:
-                     keyText += text + " "
+            case .child:
+                keyText = processChild(ids: ids, blockMap: blockMap)
             case .value:
-                for valueId in ids {
-                     let valueBlock = blockMap[valueId]
-                    guard let valueBlockType = valueBlock?.blockType else {
-                        continue
-                    }
-                    switch valueBlockType {
-                    case .word:
-
-                        if let text = valueBlock?.text {
-                        valueText += text + " "
-                        }
-                    case .selectionElement:
-                        valueSelected = keyBlock.selectionStatus == .selected ? true : false
-                    default: break
-                    }
-                }
-
+                let valueResult = processValue(ids: ids, blockMap: blockMap)
+                valueText = valueResult.0
+                valueSelected = valueResult.1
```

Comment:
Admittedly, i have not looked at this carefully enough, but it seems we are looping through a for loop and overwriting a variable keyText -- prior, we were appending text to a variable.  Is this correct?